README.md

```diff
@@ -87,9 +87,24 @@ attrs_strict._error.AttributeTypeError: names must be
 
 ### What is currently supported ?
 
-Currently there's support for simple types and types specified in the `typing` module: `List`, `Dict`, `DefaultDict`, `Set`, `Union`, `Tuple` and any combination of them. This means that you can specify nested types like `List[List[Dict[int, str]]]` and the validation would check if attribute has the specific type.
+Currently there's support for simple types and types specified in the `typing` module: `List`, `Dict`, `DefaultDict`, `Set`, `Union`, `Tuple`, `Callable` and any combination of them. This means that you can specify nested types like `List[List[Dict[int, str]]]` and the validation would check if attribute has the specific type.
 
-`Callables`, `TypeVars` or `Generics` are not supported yet but there are plans to support this in the future.
+`Callable` will validate if the callable function's annotation matches the type definition. If type does not specify any annotations then all callables will pass the validation against it.
+```python
+@attr.s
+>>>class FullyAnnonatedCallable(Callable):
```

Comment:
Does this needs to be a class, can you just make a function instead of a `class` that extends Callable? I think it would help a bit with clarity.